pages/bib/[id]/index.tsx

```diff
@@ -165,26 +224,96 @@ export default function BibPage({
                 <>
                   <Heading
                     data-testid="item-table-displaying-text"
+                    ref={itemTableHeadingRef}
                     level="h4"
                     size="heading6"
                     mb="s"
+                    tabIndex={-1}
                   >
                     {buildItemTableDisplayingString(
                       itemTablePage,
-                      bib.numPhysicalItems
+                      bib.numPhysicalItems,
+                      viewAllEnabled
                     )}
                   </Heading>
                   <ItemTable itemTableData={itemTableData} />
                 </>
               )}
-              <Pagination
-                id="bib-items-pagination"
-                initialPage={itemTablePage}
-                currentPage={itemTablePage}
-                pageCount={Math.ceil(bib.numPhysicalItems / ITEM_BATCH_SIZE)}
-                onPageChange={handlePageChange}
-                my="xl"
-              />
+              <Box display="flex" my="xl" justifyContent="space-between">
+                {!viewAllEnabled ? (
+                  <Pagination
+                    id="bib-items-pagination"
+                    initialPage={itemTablePage}
+                    currentPage={itemTablePage}
+                    pageCount={Math.ceil(
+                      bib.numPhysicalItems / ITEM_PAGINATION_BATCH_SIZE
+                    )}
+                    onPageChange={handlePageChange}
+                    width="auto"
+                  />
+                ) : null}
+                {bib.showViewAllItemsLink &&
+                  (itemsLoading && viewAllEnabled ? (
+                    <Box
+                      ml="auto"
+                      display="flex"
+                      justifyContent="flex-start"
+                      alignItems="center"
+                    >
+                      <ProgressIndicator
+                        id="bib-all-items-loading"
+                        labelText={bib.itemsViewAllLoadingMessage}
+                        size="small"
+                        indicatorType="circular"
+                        mr="xs"
+                        isIndeterminate
+                      />
+                      <Label
+                        htmlFor="bib-all-items-loading"
+                        ref={viewAllLoadingTextRef}
+                        fontSize={{
+                          base: "mobile.body.body1",
+                          md: "desktop.body.body1",
+                        }}
+                        fontWeight="medium"
+                        mb={0}
+                        // Label component does not expect tabIndex prop, so we are ignoring the typescript error that pops up.
```

Comment:
I made it a label since the ProgressIndicator label is not rendered when the size is "small". Otherwise I could probably override the styling, maybe we can fix this on the DS label as a follow up.

Reply:
How does it look like when it's regular size? I would prefer to update this on the DS side.